docs/source/en/model_doc/clip.md

```diff
@@ -14,221 +14,149 @@ rendered properly in your Markdown viewer.
 
-First, make sure to install the latest version of Flash Attention 2.
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+image = Image.open(requests.get(url, stream=True).raw)
+labels = ["a photo of a cat", "a photo of a dog", "a photo of a car"]
 
-```bash
-pip install -U flash-attn --no-build-isolation
+similarity = clip(image, candidate_labels=labels)
+print(similarity)
 ```
 
-Make also sure that you have a hardware that is compatible with Flash-Attention 2. Read more about it in the official documentation of flash-attn repository. Make also sure to load your model in half-precision (e.g. `torch.float16`)
-
-<Tip warning={true}>
-
-For small batch sizes, you might notice a slowdown in your model when using flash attention. Refer to the section [Expected speedups with Flash Attention and SDPA](#Expected-speedups-with-Flash-Attention-and-SDPA) below and select an appropriate attention implementation.
-
-</Tip>
-
-To load and run a model using Flash Attention 2, refer to the snippet below:
-
-```python
->>> import torch
->>> import requests
->>> from PIL import Image
+</hfoption>
+<hfoption id="AutoModel">
 
->>> from transformers import CLIPProcessor, CLIPModel
+```py
+import requests
+from PIL import Image
+from transformers import AutoProcessor, AutoModel
+from PIL import Image
+import torch
 
->>> device = "cuda"
->>> torch_dtype = torch.float16
+model = AutoModel.from_pretrained("openai/clip-vit-base-patch32")
+processor = AutoProcessor.from_pretrained("openai/clip-vit-base-patch32")
 
->>> model = CLIPModel.from_pretrained(
-...     "openai/clip-vit-base-patch32",
-...     attn_implementation="flash_attention_2",
-...     device_map=device,
-...     torch_dtype=torch_dtype,
-... )
->>> processor = CLIPProcessor.from_pretrained("openai/clip-vit-base-patch32")
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+image = Image.open(requests.get(url, stream=True).raw)
+text = ["a photo of a cat", "a photo of a dog", "a photo of a car"]
 
->>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
->>> image = Image.open(requests.get(url, stream=True).raw)
+inputs = processor(text=text, images=image, return_tensors="pt", padding=True)
+outputs = model(**inputs)
 
->>> inputs = processor(text=["a photo of a cat", "a photo of a dog"], images=image, return_tensors="pt", padding=True)
->>> inputs.to(device)
+image_features = outputs.image_embeds
+text_features = outputs.text_embeds
 
->>> with torch.no_grad():
-...     with torch.autocast(device):
-...         outputs = model(**inputs)
-
->>> logits_per_image = outputs.logits_per_image  # this is the image-text similarity score
->>> probs = logits_per_image.softmax(dim=1)  # we can take the softmax to get the label probabilities
->>> print(probs)
-tensor([[0.9946, 0.0052]], device='cuda:0', dtype=torch.float16)
+similarity = torch.cosine_similarity(image_features, text_features)
+print(similarity)
 ```
 
+</hfoption>
+<hfoption id="CLIPModel">
```

Comment:
Let's remove the `CLIPModel` so users can just focus on `Pipeline` and `AutoModel`